### ISSUE TITLE
Check for spurious trailing slash on RUN

### ISSUE THREAD

**Issue by colindean** [hadolint/hadolint#853]
- [ ] ~This is a bug report~
- [X] This is a feature request
- [X] I searched existing issues before opening this one

### Expected behavior

When I have something like this:
```Dockerfile
RUN something_to_run \
#   && something_else_to_run
# some
# comment
# long
# enough
# to miss
# the perilous trailing \

COPY whoops nope
```

I would expect Hadolint to warn me about the trailing slash effectively canceling the following directive or including the following directive in the RUN command.

I think the real solution here is to somehow check if the last non-whitespace character of a RUN directive is a `\` and throw a warning on that. OR, if a directive matches `\[:whitespace:](\w.*)` where the matching group is a directive keyword. I'm not familiar with the implementation of the rules so I don't beyond "look at text" how this would be implemented in Hadolint itself.

### Actual behavior

In my real-world example (below), the only indicator was this warning:

```
server/Dockerfile:16 SC2115 warning: Use "${var:?}" to ensure this never expands to / .
```

_Three_ people didn't catch the trailing slash until I did I diff against another working, similar configuration and caught the introduction of the trailing slash, likely by IDE automation.

### Steps to reproduce the behavior

My real world example:

```bash
$ hadolint --version
Haskell Dockerfile Linter 2.10.0
```

An excerpt from my real Dockerfile:

```Dockerfile
# N.b. this won't actually work in practice but it should be enough for Hadolint
FROM python:latest

RUN poetry install --no-dev --no-interaction && \
    du -sh "/root/.cache/pypoetry" && \
    rm -rf "/root/.cache/pypoetry/*" \
#    && poetry cache clear
# some
# comment
# long
# enough
# to miss
# the perilous trailing \
# THIS IS OBVIOUSLY FILLER
# The real comment is REDACTED
# but the length is important to show the difficulty of seeing what's wrong visually.
# some
# comment
# long
# enough
# to miss
# the perilous trailing \

COPY server/*.py server/config_default.yaml ${WORKDIR}/
COPY server/webapp/ ${WORKDIR}/webapp
```

**Additional environment details (OS, stack version, etc.)**

* macOS 12.4 arm64
